pybandits/pydantic_version_compact.py

```diff
@@ -0,0 +1,242 @@
+"""
+This snippet is a compatibility layer for pydantic v1 and v2.
+"""
+
+import re
+from typing import Any, Callable, Dict, Literal, Optional, Union
+from warnings import warn
+
+from pydantic import BaseModel, Field, NonNegativeFloat, PositiveInt, ValidationError, confloat, conint, constr
+from pydantic.version import version_info as _pydantic_version
+
+
+def _get_major_pydantic_version():
+    """
+    Get the major version of pydantic.
+
+    Returns
+    -------
+    major_version : str
+        The major version of pydantic.
+    """
+    full_version = re.findall(r"\d+\.\d+", _pydantic_version().split("\n")[0])[0]
+    major_version = full_version.split(".")[0]
+    return major_version
+
+
+pydantic_version = _get_major_pydantic_version()
+
+if pydantic_version == "1":
```

Comment:
You can remove this hardcoded string and declare it on top of this file with capital letters:
```
PYDANTIC_VERSION_1 == "1"
PYDANTIC_VERSION_2 == "2"
```